frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -1842,6 +1953,49 @@ class SegmentsView extends React.Component<Props, State> {
     );
   }
 
+  getExpandSubgroupsItem(groupId: number) {
+    const children: Key[] = this.getSubGroupsAsTreeNodes(groupId);
+    const expandedGroupsSet = new Set(this.state.expandedGroupKeys);
+    const areAllChildrenExpanded = children.every((childNode) => expandedGroupsSet.has(childNode));
+    const isGroupItselfExpanded = expandedGroupsSet.has(getKeyForGroupId(groupId));
+    if (areAllChildrenExpanded && isGroupItselfExpanded) {
+      return null;
+    }
+    const allExpandedGroups = children.concat(this.state.expandedGroupKeys ?? []);
```

Comment:
why `?? []`? isn't `this.state.expandedGroupKeys` always defined?

also this variable is only needed in the `onClick` handler. it can be moved there.